src/features/api-keys/lib.ts

```diff
@@ -107,7 +113,20 @@ const getKeyMetadata = (reqBody: { name?: any; description?: any }) => {
 		);
 	}
 
-	return { name, description };
+	if (
+		expiryDate != null &&
+		(typeof expiryDate !== 'string' || isNaN(new Date(expiryDate).getTime()))
+	) {
+		throw new errors.BadRequestError('Key expiry date should be ISO formatted');
```

Comment:
I think that pine supports numeric timestamps as well, so how about v ?
```suggestion
		throw new errors.BadRequestError('Key expiry date should be a valid date');
```